File: benchmarks/bench_16.go
package main

import (
	"fmt"
	"time"
)

func harmonic_sum(limit int64) float64 {
    s := 0.0
    i := int64(1)
    for i < limit {
        s += 1.0 / float64(i)
        i += 1
    }
    return s
}

func main() {
    t1 := time.Now()
    res := harmonic_sum(2000000)
    t2 := time.Now()
    fmt.Println(res)
    fmt.Printf("%.6f\n", t2.Sub(t1).Seconds())
}

File: benchmarks/bench_16.py
import time
import sys
sys.setrecursionlimit(200000)

def harmonic_sum(limit):
    s = 0.0
    i = 1
    while i < limit:
        s += 1.0 / i
        i += 1
    return s


t1 = time.perf_counter()
res = harmonic_sum(2000000)
t2 = time.perf_counter()
print(res)
print(f"{t2 - t1:.6f}")
